### ISSUE TITLE
Records: warn if either Equals or GetHashCode is redefined, but not both

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#47513]
If you redefine one method, the other gets immediately out of sync and you don't get any warning.
```cs
using System;
public sealed record C {
    public object Data;
    public bool Equals(C c) { return false; }
   // Compiler generated GetHashCode considers Data and is not in agreement with Equals
}
```

**Comment by alrz** [comment]
To be clear, I don't think a special attribute for this purpose is the best approach, as long as there exist *any* workaround except redefining those methods.

**Comment by Youssef1313** [comment]
@jaredpar Can we close this? it was addressed in #47515 but wasn't auto-closed since the PR didn't target the default branch.

**Comment by Youssef1313** [comment]
Pinging @jaredpar. This can be closed now I think.
